tests/model/test_utils.py

```diff
@@ -41,3 +43,39 @@
 )
 def test_stack_delete(template_url, bucket, path):
     assert extract_bucket_name_and_path_from_url(template_url) == (bucket, path)
+
+
+@pytest.mark.parametrize(
+    "input, expected_output",
+    [
+        ("", ""),
+        (" ", " "),
+        ("\r\n", "\r\n"),
+        ("\t", "  "),
+        ("a\t", "a  "),
+        ("\ta", "  a"),
+        ("\ta\t", "  a  "),
+        ("a\ta", "a  a"),
+    ],
+)
+def test_replace_tabs(input, expected_output):
+    assert replace_tabs(input) == expected_output
+
+
+@pytest.mark.parametrize(
+    "input, expected_output",
+    [
+        ('{"A": "a"}', {"A": "a"}),
+        ("A: a", {"A": "a"}),
+        ("\tA: a", {"A": "a"}),
+        ("\tA:\t  a\n  B: b", {"A": "a", "B": "b"}),
+    ],
+)
+def test_convert_json_or_yaml_to_dict(input, expected_output):
+    assert convert_json_or_yaml_to_dict(input) == expected_output
+
+
+def test_convert_json_or_yaml_to_dict_yaml_with_tabs():
+    cf_path = "tests/test_templates/others/yaml_with_tabs.yml"
```

Comment:
Use Pathlib, setting the current filename as source, otherwise calling this test from somewhere different than the project root will cause this test to fail